lib/auth0/mixins/initializer.rb

```diff
@@ -4,21 +4,21 @@ module Mixins
     module Initializer
       # Default initialization mechanism, moved here to keep Auth0::Client clear
       # accepts hash as parameter
-      # you can get all required fields from here: https://auth0.com/docs/auth-api
+      # you can get all required fields from here:
+      #  https://auth0.com/docs/auth-api
       #
-      # To run using api v2, pass api_version: 2 when creating a client
+      # By Default API v2
       def initialize(config)
-        options = Hash[config.map{|(k,v)| [k.to_sym,v]}]
-        domain = api_domain options
-        raise InvalidApiNamespace, "Api namespace must supply an API domain" if domain.nil?
-        self.class.base_uri "https://#{domain}"
+        options = Hash[config.map { |(k, v)| [k.to_sym, v] }]
+        self.class.base_uri base_url(options)
         self.class.headers client_headers(config)
-        self.extend Auth0::Api::AuthenticationEndpoints
-        @client_id      = options[:client_id]
-        initialize_v2(options) if api_v2?(options)
-        initialize_v1(options) if api_v1?(options)
-        raise InvalidCredentials, "Must supply a valid API token" if @token.nil?
-        self.class.headers "Authorization" => "Bearer #{@token}"
+        extend Auth0::Api::AuthenticationEndpoints
+        @client_id = options[:client_id]
+        if api_v1?(options)
+          initialize_v1(options)
+        else
+          initialize_v2(options)
+        end
       end
```

Comment:
That would break rubocop rule of branch condition size (not sure why :s)
***C: Assignment Branch Condition size for initialize is too high. [15.17/15]***